### DISCUSSION TITLE
One repository (synopackage.com) => many sources

### DISCUSSION THREAD

**Discussion by szyb** [szyb/synopackage_dotnet#160] [Ideas]
I'm thinking of synopackage repository which will return packages from all active repositories (or defined by user, or just recommended i.e. top 5). This is technically possibile and I'm wondering if this would be useful or helpful for the community.

The goal is to act as a proxy and provide data from origin server without any change of its content. The links to SPKs will also lead to origin servers (caching packages requires a lot of space which we don't have).

I'm wondering what do you think about this and what would you expect from such a repository?

**Comment by vletroye** [comment]
That would be really really great. But I am wondering if issues couldn't  occur... 
Wouldn't you have conflicts between various flavor of a package if by nay chance several publishers did customize it without any renaming ?

Ex.: I just copied the package "Git" to create a custom version for DSM 7.x (for my own purpose).. But didn't change the name before publishing on my own SSPKS...

**Reply by szyb** [reply]
Good point. There could be also the same package name across few servers, but in different version. Need to check this scenario as well. 
Thanks for the feedback.

**Comment by szyb** [comment]
It's been a while since original post. Almost a year 😮 If anyone is interested in this feature I can say that I've created a little POC (Proof Of Concept) and it works fine :)
There is some work to do (together with refactor the code), but keep your finger crossed :) If everything goes well I will publish private beta testing to see how it works on different versions and models. 
Below: successfully added repository (containing all packages from _synocommunity_ and _cphub_)
![image](https://user-images.githubusercontent.com/1598139/145276859-a93db58f-dd27-4be5-812f-e82e4ed53ae9.png)

**Comment by szyb** [comment]
EDIT: private testing is over

Ok, so here we go :) For now only for testing and it's not public. Just only for those willing.
Please sing up to the beta tests using this link below

https://forms.gle/UnVVeBrPqCda64CeA 

You will get a personalized link which you shall not share with anyone. If you want to share, please share the sing up from link ;) 
Also if you plan to test the repository on multiple devices please request multiple links.

Enjoy!

**Reply by szyb** [reply]
FYI: Synopackage Proxy Repository is online and ready to use for everyone (since 07.03.2022)
More details: https://synopackage.com/repository

**Reply by burghy86** [reply]
great job man!! I had missed this discussion. congratulations for your tenacity

**Reply by szyb** [reply]
Thanks :)
BTW: Let me ask you something about synologyitalia: does it support DSM 7.x? I was thinking to add it to SynopackageChoice repository channel but I was not sure whether there are packages to DSM7

**Reply by burghy86** [reply]
I hear the developer and tell you
